code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,274 @@
+	force = 10
+	origin_tech = null
+	selfcharge = 1
+	var/inawaymission = 1
+
+/obj/item/weapon/gun/energy/laser/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == (MAX_Z + 1))
+		if(inawaymission)
+			..()
+		else
+			to_chat(get(src, /mob), "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
+			inawaymission = 1
+	else
+		if(inawaymission)
+			to_chat(get(src, /mob), "<span class='danger'>Your [src] deactivates, as it is out of range from its power source.</span>")
+			power_supply.charge = 0
+			inawaymission = 0
+			update_icon()
+
+
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin
+	name = "beaker 'Black Terror Venom'"
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin/New()
+	..()
+	reagents.add_reagent("terror_black_toxin", 50)
+	update_icon()
+
+
+/obj/item/weapon/gun/energy/decloner/awaymission_spidergun
+	name = "Modified Decloner"
+	desc = "A modified biological decloner, tweaked to destroy terror spiders more effectively."
+	icon_state = "decloner"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	ammo_type = list(/obj/item/ammo_casing/energy/declone_spider)
+	can_charge = 0
+
+/obj/item/ammo_casing/energy/declone_spider
+	projectile_type = /obj/item/projectile/energy/declone/declone_spider
+	select_name = "declone"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+
+/obj/item/projectile/energy/declone/declone_spider
+	name = "declone"
+	icon_state = "declone"
+	damage = 20
+	damage_type = CLONE
+
+/obj/machinery/computer/id_upgrader
+	name = "ID Upgrade Machine"
+	icon_state = "guest"
+	icon_screen = "pass"
+	var/list/access_to_give = list(271)
+	var/beenused = 0
+	var/door_to_open = "UO71_Start"
+
+/obj/machinery/computer/id_upgrader/attackby(obj/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/card/id))
+		var/obj/item/weapon/card/id/I = O
+		if(!access_to_give.len)
+			to_chat(user, "<span class='notice'>This machine appears to be configured incorrectly.</span>")
+			return
+		var/did_upgrade = 0
+		for(var/this_access in access_to_give)
+			if(!(this_access in I.GetAccess()))
+				// don't have it - add it
+				I.access |= this_access
+				did_upgrade = 1
+		if(did_upgrade)
+			to_chat(user, "<span class='notice'>An access type was added to your ID card.</span>")
+			if(!beenused)
+				spawn(1)
+					beenused = 1
+					var/unlocked_something = 0
+					for(var/obj/machinery/door/poddoor/P in airlocks)
+						if(P.density && P.id_tag == door_to_open && P.z == z)
+							P.open()
+							unlocked_something = 1
+					if(unlocked_something)
+						to_chat(user, "<span class='danger'>Activating the machine has unlocked a way forward!</span>")
+		else
+			to_chat(user, "<span class='notice'>Your ID card already has all the access this machine can give.</span>")
+	else
+		to_chat(user, "<span class='notice'>Use an ID card on the [src] instead.</span>")
```

Comment:
don't give an error message if they hit it with something else, they can work out what an `ID upgrader` does on their own